src/conversations/Conversations.ts

```diff
@@ -61,33 +67,46 @@ export class ConversationCache {
  */
 export default class Conversations {
   private client: Client
+  private v1Cache: ConversationCache
   private v2Cache: ConversationCache
 
   constructor(client: Client) {
     this.client = client
+    this.v1Cache = new ConversationCache()
     this.v2Cache = new ConversationCache()
   }
 
   /**
    * List all conversations with the current wallet found in the network, deduped by peer address
    */
   async list(): Promise<Conversation[]> {
-    const [seenPeers, v2Convos] = await Promise.all([
-      this.getIntroductionPeers(),
+    const [v1Convos, v2Convos] = await Promise.all([
+      this.listV1Conversations(),
       this.listV2Conversations(),
     ])
 
-    const v1Convos: Conversation[] = []
-    seenPeers.forEach((sent, peerAddress) =>
-      v1Convos.push(new ConversationV1(this.client, peerAddress, sent))
-    )
-
     const conversations = v1Convos.concat(v2Convos)
 
     conversations.sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime())
     return conversations
   }
 
+  private async listV1Conversations(): Promise<Conversation[]> {
+    return this.v1Cache.load(async ({ latestSeen }) => {
+      const seenPeers = await this.getIntroductionPeers({
+        startTime: latestSeen
+          ? new Date(+latestSeen - CLOCK_SKEW_OFFSET_MS)
+          : undefined,
+        direction: SortDirection.SORT_DIRECTION_ASCENDING,
+      })
+
+      return Array.from(seenPeers).map(
+        ([peerAddress, sent]) =>
+          new ConversationV1(this.client, peerAddress, sent)
+      )
+    })
+  }
+
   private async listV2Conversations(): Promise<Conversation[]> {
     return this.v2Cache.load(async ({ latestSeen }) => {
```

Comment:
Just thinking out loud here, I wonder if there's a way for someone to only want to cache a few conversations (ones that they saw come thru on pushes or something). Should we indicate/document that the cache must be a complete cache or else anything older than the most recent will be dropped?

Reply:
Yes. I'll write up a section and add to the README

Reply:
Added a section to the README and included some code comments